target_chains/ton/contracts/contracts/Pyth.fc

```diff
@@ -1,10 +1,36 @@
 #include "imports/stdlib.fc";
 #include "common/errors.fc";
 #include "common/storage.fc";
+#include "common/utils.fc";
+#include "./Wormhole.fc";
+
+const int ACCUMULATOR_MAGIC = 0x504e4155; ;; "PNAU" (Pyth Network Accumulator Update)
+const int MAJOR_VERSION = 1;
+const int MINIMUM_ALLOWED_MINOR_VERSION = 0;
+
+slice verify(slice data) {
+    int magic = data~load_uint(32);
+    throw_unless(ERROR_INVALID_MAGIC, magic == ACCUMULATOR_MAGIC);
+    int major_version = data~load_uint(8);
+    throw_unless(ERROR_INVALID_MAJOR_VERSION, major_version == MAJOR_VERSION);
+    int minor_version = data~load_uint(8);
+    throw_if(ERROR_INVALID_MINOR_VERSION, minor_version < MINIMUM_ALLOWED_MINOR_VERSION);
+    int trailing_header_size = data~load_uint(8);
+    ;; skip trailing headers and update type (uint8)
+    data~skip_bits(trailing_header_size);
+    data~skip_bits(8);
```

Comment:
what is this that you are skipping? It should be the update type right? In that case comment it. Please not that later you need to verify that update type is what we support and if this function is only about to verify the header, this should be in another function. You can leave a TODO note and later move it out.

Reply:
its already commented on line 19